server_cmds.go

```diff
@@ -165,13 +165,27 @@ func (s *ServerMethods) ExtendedList(options ...string) ([]*Server, error) {
 	var servers []*Server
 	var outputServers []*Server
 
-	var latestServerID = s.LatestServerID
-	var latestServerByID = s.LatestServerByID
+	var latestServerPort int
 
 	if _, err := s.ExecCmd(NewCmd("serverlist").WithOptions(options...).WithResponse(&servers)); err != nil {
 		return nil, err
 	}
 
+	info, err := s.Whoami()
+	if err != nil {
+		return nil, err
+	}
+
+	latestServerPort = info.SelectedServerPort
+
+	defer func() {
+		if err := s.UsePort(latestServerPort); err != nil {
+			// Can't handle an error in this case because the
```

Comment:
This is not true you just need to declare err in the function definition so it can be set during the defer.

Note you'll only want to set err if its nil so you'll need an err2 for the initial capture.

Reply:
Hi @thannaske any update on this?

Reply:
Hey, sorry for the late response. I'm currently on vacation and will head back to the issue soon.